cache/src/lib.rs

```diff
@@ -287,17 +402,19 @@ impl Cache {
 	pub async fn get_or_create(
 		&self,
 		accessible: &AccessibleProxy<'_>,
+		arc: Arc<Self>,
```

Comment:
This would be cleaner as a static method `get_or_create(self: Arc<Self>, accessible: &AccessibleProxy<'_>)`